### DISCUSSION TITLE
Add PHP 8.3 `#[Override]` attribute to methods

### DISCUSSION THREAD

**Discussion by hockdudu** [PHP-CS-Fixer/PHP-CS-Fixer#7911] [Q&A] [answered]
Hello,

PHP 8.3 introduced the `#[\Override]` attribute (https://php.watch/versions/8.3/override-attr).

This attribute can be added to methods to ensure the method is overriding a method from a parent class or interface.

Now, I'd like to enforce it on my project, but I can't find any setting or tool to do this automatically.

Is this something in the scope of CS Fixer? If not, is there another tool that has this feature?

I found that Psalm has an option to throw an error if the attribute is not present (https://psalm.dev/docs/running_psalm/configuration/#ensureoverrideattribute), but I couldn't find anything that automatically adds this attribute to methods.

Thank you 😊

**Comment by Wirone** [comment, marked as answer]
PHP-CS-Fixer does not have an inheritance context, so it can't determine such methods. Luckily there's a [Rector rule](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#addoverrideattributetooverriddenmethodsrector) for that 🙂.

**Reply by hockdudu** [reply]
I didn't know about that Rector rule, thanks! 🤗
